### PR TITLE
Slack - code cleanup & add missing validation

### PR DESCRIPTION
**Pull request by dalyd14**
## Description of the change

> code cleanup, add validation, and improve test cases

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by akashrpo** [comment]
@dalyd14 Let's also put a validation for `message.type` as it will throw an error `Cannot read property 'toLowerCase' of undefined` when not declared.

**Review by utsabc — APPROVED** [review]

**Review by akashrpo — APPROVED** [review]

File: v0/destinations/slack/transform.js [hunk 1]
```diff
@@ -305,6 +307,12 @@ function process(event) {
   const respList = [];
   let response;
   const { message, destination } = event;
+  if (!message.type) {
```
Comment (saikumarrs) [inline v0/destinations/slack/transform.js:310]: [Non-blocking comment] Can we check if events without type get/should be filtered at the gateway/server level?
@akashrpo

File: v0/destinations/slack/transform.js [hunk 2]
```diff
@@ -178,6 +178,9 @@ function processTrack(message, destination) {
   const eventChannelConfig = destination.Config.eventChannelSettings;
   const eventTemplateConfig = destination.Config.eventTemplateSettings;
 
+  if (!message.event) {
```
Comment (saikumarrs) [inline v0/destinations/slack/transform.js:181]: Please change the PR title to something meaningful.

**Review by saikumarrs — APPROVED** [review]

File: v0/destinations/slack/transform.js (hunk 1 above)
Comment (akashrpo) [reply v0/destinations/slack/transform.js:310]: It will throw an error as - Cannot read property 'toLowerCase' of undefined

**Pull request merged**